dataikuapi/dss/managedfolder.py

```diff
@@ -90,18 +90,18 @@ def delete_file(self, path):
         return self.client._perform_empty(
                 "DELETE", "/projects/%s/managedfolders/%s/contents/%s" % (self.project_key, self.odb_id, path))
 
-    def put_file(self, name, f):
+    def put_file(self, path, f):
         """
         Upload the file to the managed folder
         
         Args:
             f: the file contents, as a stream
-            name: the name of the file
+            path: the path of the file
         """
 
         return self.client._perform_json_upload(
-                "POST", "/projects/%s/managedfolders/%s/contents/" % (self.project_key, self.odb_id),
-                name, f)
+                "POST", "/projects/%s/managedfolders/%s/contents/%s" % (self.project_key, self.odb_id, path),
```

Comment:
1. Appending raw path to the URL won't always work because of URL encoding/decoding rules. For instance, if the filename contains a `%` character it will either fail, or interpret it like an encoded character (eg. `%20` will become a space, etc)
2. This function raises `UnicodeEncodeError: 'ascii' codec can't encode characters` if non-ascii characters are present in the path


You may want to use a function like `requests.utils.quote(...)` to fix (1). Luckily, it will also solve (2) at the same time because special characters will be replaced by percent-encoded entities